js/ai/src/retriever.ts

```diff
@@ -294,3 +294,106 @@ export function indexerRef<
   return { ...options };
 }
+
+function itemToDocument<R>(
+  item: any,
+  options: SimpleRetrieverOptions
+): Document {
+  if (!item)
+    throw new GenkitError({
+      status: 'INVALID_ARGUMENT',
+      message: `Items returned from simple retriever must be non-null.`,
+    });
+  if (typeof item === 'string') return Document.fromText(item);
+  if (typeof options.content === 'function') {
+    const transformed = options.content(item);
+    return typeof transformed === 'string'
+      ? Document.fromText(transformed)
+      : new Document({ content: transformed });
+  }
+  if (typeof options.content === 'string' && typeof item === 'object')
+    return Document.fromText(item[options.content]);
+  throw new GenkitError({
+    status: 'INVALID_ARGUMENT',
+    message: `Cannot convert item to document without content option. Item: ${JSON.stringify(item)}`,
+  });
+}
+
+function itemToMetadata(
+  item: any,
+  options: SimpleRetrieverOptions
+): Document['metadata'] {
+  if (typeof item === 'string') return undefined;
+  if (Array.isArray(options.metadata) && typeof item === 'object') {
+    const out: Record<string, any> = {};
+    options.metadata.forEach((key) => (out[key] = item[key]));
+  }
+  if (typeof options.metadata === 'function') return options.metadata(item);
+  if (!options.metadata && typeof item === 'object') {
+    const out = { ...item };
+    if (typeof options.content === 'string') delete out[options.content];
+    return out;
+  }
+  throw new GenkitError({
+    status: 'INVALID_ARGUMENT',
+    message: `Unable to extract metadata from item with supplied options. Item: ${JSON.stringify(item)}`,
+  });
+}
+
+export interface SimpleRetrieverOptions<
+  C extends z.ZodTypeAny = z.ZodTypeAny,
+  R = any,
+> {
+  /** The name of the retriever you're creating. */
+  name: string;
+  /** A Zod schema containing any configuration info available beyond the query. */
+  configSchema?: C;
+  /**
+   * Specifies how to extract content from the returned items.
+   *
+   * - If a string, specifies the key of the returned item to extract as content.
+   * - If a function, allows you to extract content as text or a document part.
+   **/
+  content?: string | ((item: R) => Document['content'] | string);
+  /**
+   * Specifies how to extract metadata from the returned items.
+   *
+   * - If an array of strings, specifies list of keys to extract from returned objects.
+   * - If a function, allows you to use custom behavior to extract metadata from returned items.
+   */
+  metadata?: string[] | ((item: R) => Document['metadata']);
+}
+
+/**
+ * defineSimpleRetriever makes it easy to map existing data into documents that
+ * can be used for prompt augmentation.
+ *
+ * @param options Configuration options for the retriever.
+ * @param handler A function that queries a datastore and returns items from which to extract documents.
+ * @returns A Genkit retriever.
+ */
+export function defineSimpleRetriever<
```

Comment:
This is neat and I think it should exist, just wondering if it should exist here... the more I think about it the more I think it's probably OK that it's exported via `@genkit-ai/ai/retriever`... not sure, just saying it out loud...  feel free to ignore.

Reply:
I think it's the right place for it, but fair to ask.